src/test/java/com/faforever/api/clan/ClanControllerTest.java

```diff
@@ -0,0 +1,284 @@
+package com.faforever.api.clan;
+
+import com.faforever.api.client.ClientType;
+import com.faforever.api.client.OAuthClient;
+import com.faforever.api.client.OAuthClientRepository;
+import com.faforever.api.data.domain.Clan;
+import com.faforever.api.data.domain.ClanMembership;
+import com.faforever.api.data.domain.Player;
+import com.faforever.api.data.domain.User;
+import com.faforever.api.player.PlayerRepository;
+import com.faforever.api.user.UserRepository;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.authentication.encoding.ShaPasswordEncoder;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.Base64Utils;
+import org.springframework.web.context.WebApplicationContext;
+
+import javax.inject.Inject;
+import javax.servlet.Filter;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ClanControllerTest {
```

Comment:
`@IntegrationTest` is deprecated
![image](https://cloud.githubusercontent.com/assets/1217999/23622670/d71eb5f6-029f-11e7-91c6-fa6b90a46764.png)